lesson_02_homework.rb

```diff
@@ -1,46 +1,154 @@
 # Code Reading
 
+# Add a comment (a line starting with `#` ) before each line explaining what it does. 
+# Add a comment before the method name explaining, in plain english, what the method does. 
+# Try copying and pasting this into a ruby file and modifying it to help you figure it out. 
+# Try adding "puts" lines in at different places to see what happens.
+
+
+# ```ruby
+# defining a method (with 3 parameters): first name, last name, and title using string interpolation
 def full_name(first_name, last_name, title)
-  # Example comment line
 
+# name starts out with no value
   name = nil
 
+# if we have parameters title and first_name and last_name
   if title && first_name && last_name
+# pass values/variables that are assigned for these parameters
+# variable for title is "Mr.", variable for first_name is "Simon", variable for last_name is "Fuzzyface"
+# then name = Mr. Simon Fuzzyface
     name = title + " " + first_name + " " + last_name
+
+# unless the parameters passed are for title and last_name
+# then name = Mr. Fuzzyface
   elsif title && last_name
     name = title + " " + last_name
+
+# unless the parameters passed are for first_name and last_name
+# then name = Simon Fuzzyface
   elsif first_name && last_name
     name = first_name + " " + last_name
+
+# unless the parameters passed are for first_name
+# then name = Simon
   elsif first_name
     name = first_name
+
+# unless first_name or last_name are not passed
+# raise is used as default for when the necessary variables are not passed, 
+# so we get "Oh no, that doesn't look like a name"
   else
     raise "Oh no, that doesn't look like a name"
   end
-
+# the method has put the values into the appropriate variables
   return name
 end
 
-# Fix Broken Code
+puts full_name("Simon", "Fuzzyface", "Mr.")
+puts full_name("Mr.", "Fuzzyface", "")
```

Comment:
I'm not understanding your correction...where am I missing "nil"? Can we go over this in class at break?
Thank you!